### PR TITLE
Responses

### PR DESCRIPTION
**Pull request by Hellowlol**
No ready for merge, but it's getting closer.

### CODE REVIEW

**Comment by Hellowlol** [comment]
Note to self:
- remove all xfail before merge.  Dunno how to handle this, its seems its methods that are inherited but itsnt supported for that mediatype. I only added xfail to reduce the noise.

File: plexapi/video.py [hunk 1]
```diff
@@ -381,7 +430,11 @@ def __repr__(self):
         clsname = self.__class__.__name__
         key = self.key.replace('/library/metadata/', '').replace('/children', '') if self.key else 'NA'
         title = self.title.replace(' ', '.')[0:20].encode('utf8')
-        return '<%s:%s:%s:%s>' % (clsname, key, self.grandparentTitle, title)
+        return '<%s:%s:%s:%s>' % (clsname, key, self.parentTitle, title)
+
+    def download(self, savepath=None, keep_orginal_name=False, **kwargs):
+        for ep in self.episodes():
```
Comment (Hellowlol) [inline plexapi/video.py]: fix me

**Comment by Hellowlol** [comment]
https://coveralls.io/github/Hellowlol/plexapi
https://travis-ci.org/Hellowlol/plexapi

**Comment by Hellowlol** [comment]
So coverage is at 82% atm. But there is still alof of low hanging fruit. I just need figure out a way to test the clients.

**Comment by mjs7231** [comment]
Maybe it's worth resolving these conflicts now.  If there are not that many changes to the core plexapi, it's probably not going to be too intrusive to get it into master sooner than later.

**Comment by Hellowlol** [comment]
Yeah, but I'd prefer to just squash the entire thing and so I don't have to keep resolving the conflicts over and over. While the pr will huge I'm pretty confident it will work out one i since there should be tests for everything except for actions that require a client.

**Comment by mjs7231** [comment]
If you resolve the conflicts now, I'd merge it now and continue the documentation work with these changes merged in master.

**Comment by Hellowlol** [comment]
I don't have time today. I'll do it tomorrow.

**Comment by mjs7231** [comment]
No rush, if I get time tonight I'll take a stab at it as well.

**Comment by Hellowlol** [comment]
The tests fail because the enviroments PLEX_TEST_USERNAME, PLEX_TEST_TOKEN, PLEX_TEST_PASSWORD isnt exported. I have added it, but it dont seems to take. Maybe you need to do that yourself. Some of the tests are abit flaky add will fail for time to time when running on travis. This is because there concurrent calls to pms editing the watched status at the same time. Just restart the tests that fail.

**Pull request merged**